lib/exnumerator.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Exnumerator do
+  defmacro __using__(opts) do
+    quote location: :keep, bind_quoted: [values: opts[:values]] do
+      @behaviour Ecto.Type
+
+      @doc """
+      Returns all possible values for the enum.
+      """
+      def values, do: unquote(values)
+
+      @doc """
+      Returns a random value for the enum.
+      """
+      def sample, do: unquote(values) |> Enum.random()
+
+      @impl true
+      def type, do: :string
+
+      @impl true
+      def cast(term)
+
+      @impl true
+      def dump(term)
+
+      @impl true
+      def load(term)
+
+      cond do
+        Keyword.keyword?(values) ->
```

Comment:
@rzane @KamilLelonek maybe what we should do is to remove as much macros as we could?

Help me out here because I am not really good at this macros things 😄 

```elixir
quote do
  def cast(value), do: handle_cast(unquote(values), value)
  def dumb(value), do: handle_dumb(unquote(values), value)
  def load(data), do: handle_data(unquote(values), data)
end
```

Reply:
Let me see, BTW please open Elixir Slack :)